core/Plugin/Archiver.php

```diff
@@ -83,6 +87,83 @@ public function __construct(ArchiveProcessor $processor)
         $this->enabled = true;
     }
 
+    private function getPluginName()
+    {
+        // TODO: consider extracting to a reusable method or a trait, or use another approach to getting plugin's name
+        $className = get_class($this);
+        $parts = explode('\\', $className);
+        $plugin = $parts[2];
+        return $plugin;
+    }
+
+    /**
+     * @return ArchiveProcessor\RecordBuilder[]
+     * @throws \DI\DependencyException
+     * @throws \DI\NotFoundException
+     */
+    private function getRecordBuilders()
```

Comment:
Even though this returns an array of objects, we could at least type hint array as return.
```suggestion
    private function getRecordBuilders(): array
```